dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java

```diff
@@ -189,21 +188,29 @@ public void destroy() {
 
     @Override
     public synchronized void notify(List<URL> urls) {
-        List<URL> categoryUrls = urls.stream()
+        Map<String, List<URL>> categoryUrls = urls.stream()

```

Comment:
could you add filter for url not null as well. In case if caller is providing null url, this is just to be safer. In case if you are already doing then ignore my comment.

Reply:
@khanimteyaz done